### DISCUSSION TITLE
Where does the default ohmyzsh install get its URL completions from?

### DISCUSSION THREAD

**Discussion by humdogm** [ohmyzsh/ohmyzsh#10230] [Help] [answered]
Hi, I'm using ohmyzsh on WSL. Recently when using the autocomplete, I saw a url `media.opencandy.com`. I went to look up what it is and apparently it's a url used by a piece of known adware. When I checked the rest of the url auto completes by typing `curl <TAB>` I saw all the below completions, most of which look like adware. Where does the default ohmyzsh install find urls to autocomplete?

![image](https://user-images.githubusercontent.com/6947043/135161941-188594f0-e200-4081-b17d-b51d97e28e1b.png)

**Comment by mcornella** [comment, marked as answer]
It looks like you used some `/etc/hosts`-based adblocker which adds all these domains to `/etc/hosts` to block connections to them. The completion function for hosts uses `getent hosts` or parses the `/etc/hosts` command, so they will appear in your completion results.

**Reply by humdogm** [reply]
Yeah, looks like Unchecky changes the Windows hosts file, and those changes came to the WSL as well. Would it be possible to filter out entries from `getent hosts` that start with 0.0.0.0? Those probably shouldn't autocomplete when those are inaccessible over the network.
